File: fsharp/FsFind/FsFindLib/Logger.fs
namespace FsFindLib

open System

module Logger = 
    let Log (msg : string) : unit =
        printfn $"%s{msg}"

    let LogErrorColor (msg : string) (colorize : bool) : unit =
        let err =
            if colorize then $"%s{Color.BoldRed}ERROR: %s{msg}%s{Color.Reset}"
            else $"ERROR: %s{msg}"
        Console.Error.WriteLine err

    let LogError (msg : string) : unit =
        LogErrorColor msg true

;;

File: csharp/CsFind/CsFindLib/Logger.cs
using System;

namespace CsFindLib;

public static class Logger
{
	public static void Log(string message)
	{
		Console.WriteLine(message);
	}

	public static void LogError(string message, bool colorize = true)
	{
		var err = colorize
			? $"{ConsoleColor.BoldRed}ERROR: {message}{ConsoleColor.Reset}"
			: $"ERROR: {message}";
		Console.Error.WriteLine(err);
	}
}
